docusaurus/docs/dev-docs/preview.md

```diff
@@ -0,0 +1,305 @@
+- or using a different domain for previews like `preview.mysite.com/your-path`.
+
+When [Draft & Publish](/user-docs/content-manager/saving-and-publishing-content.md) is enabled for your content-type, you can also directly leverage Strapi's `status` parameter to handle the logic within the Preview handler, using the following generic approach:
+
+```javascript
+async handler(uid, { documentId, locale, status }) {
+   const document = await strapi.documents(uid).findOne({ documentId });
+   const pathname = getPreviewPathname(uid, { locale, document });
+   if (status === 'published')  { 
+      // return the published version
+   }
+   // return the draft version
+},
+```
+
+A more detailed example using the draft mode of Next.js is given in the [basic implementation guide](#3-add-handler-logic).
+
+## Basic implementation guide
+
+Follow these steps to add Preview capabilities to your content types.
+
+### 1. Create the Preview configuration
+
+Create a new file `/config/admin.ts` (or update it if it exists) with the following basic structure:
+
+```javascript title="config/admin.ts"
+export default ({ env }) => ({
+  // Other admin-related configurations go here
+  // (see docs.strapi.io/dev-docs/configurations/admin-panel)
+  preview: {
+    enabled: true,
+    config: {
+      allowedOrigins: env('CLIENT_URL'),
+      async handler (uid, { documentId, locale, status }) => {
+        // Handler implementation coming in step 3
+      },
+    },
+  },
+});
+```
+
+### 2. Add URL generation logic
+
+Add the URL generation logic with a `getPreviewPathname` function. The following example is taken from the [Launchpad](https://github.com/strapi/LaunchPad/tree/feat/preview) Strapi demo application:
+
+```typescript title="config/admin.ts"
+// Function to generate preview pathname based on content type and document
+const getPreviewPathname = (uid, { locale, document }): string => {
+  const { slug } = document;
+  
+  // Handle different content types with their specific URL patterns
+  switch (uid) {
+    // Handle pages with predefined routes
+    case "api::page.page":
+      switch (slug) {
+        case "homepage":
+          return `/${locale}`; // Localized homepage
+        case "pricing":
+          return "/pricing"; // Pricing page
+        case "contact":
+          return "/contact"; // Contact page
+        case "faq":
+          return "/faq"; // FAQ page
+      }
+    // Handle product pages
+    case "api::product.product": {
+      if (!slug) {
+        return "/products"; // Products listing page
+      }
+      return `/products/${slug}`; // Individual product page
+    }
+    // Handle blog articles
+    case "api::article.article": {
+      if (!slug) {
+        return "/blog"; // Blog listing page
+      }
+      return `/blog/${slug}`; // Individual article page
+    }
+  }
+  return "/"; // Default fallback route
```

Comment:
This is actually something we don't recommend, we're about to remove it in the launchpad code.

It's okay for some content types to not have a preview if it doesn't make sense. A Global single type with some site metadata for example will not have a matching frontend page. In these cases, the handler function should return null, and the preview UI will not be shown in the URL. This is how you enable/disable preview per content type.

I can ping you when we have the launchpad fix PR if you're interested

Reply:
here's the commit https://github.com/strapi/LaunchPad/pull/19/commits/6ef83bc625eabe994aa48bf3b8686af0cd17fd6b

Reply:
Super helpful, thank you! I added a note with your text in the docs because it was crystal clear! :chefkiss: